src/sdk/block-engine/searcher.ts

```diff
@@ -125,6 +125,35 @@ export class SearcherClient {
     });
   }
 
+  // Yields on account updates owned by the provided list of programs.
+  async *programUpdates(
+    programs: PublicKey[],
+    onError: (e: Error) => void
+  ): AsyncGenerator<VersionedTransaction[]> {
+    const stream: ClientReadableStream<PendingTxNotification> =
+      this.client.subscribeMempool({
+        programV0Sub: {
+          programs: programs.map(p => p.toString()),
+        },
+      });
+
+    stream.on('error', e => {
+      onError(e);
+    });
+
+    for await (const pendingTxNotification of stream) {
+      try {
+        yield deserializeTransactions(pendingTxNotification.transactions);
+      } catch (e) {
+        if (e instanceof Error) {
+          onError(e);
+        } else {
+          onError(new Error('Deserialization error'));
```

Comment:
can you log e here?

Reply:
done. also removed this try/catch on bundleResults as it would never be reached